src/requests/typings/function.ts

```diff
@@ -45,6 +48,25 @@ export type GraphQLQueryResponse = {
   extensions?: Record<string, unknown>
 }
 
+type AppEventFilter<T> = {
+  type: typeof APP_EVENT_FILTER
+  entityProps: T
+  entityAction: string // 'create' | 'publish' etc etc.
+}
+// TODO: add all of the other app event subscription entities and topics/actions
+export type AppEventEntryFilter = {
+  entityType: 'Entry'
+} & AppEventFilter<EntryProps>
+export type AppEventAssetFilter = {
+  entityType: 'Asset'
+} & AppEventFilter<AssetProps>
+export type AppEventFilterRequest = AppEventEntryFilter | AppEventAssetFilter
+
+export type AppEventFilterResponse = {
+  result: boolean
+  errors?: readonly Record<string, any>[]
```

Comment:
What is the intent of this property?

Reply:
we were following suit with delivery function response shape in case there were errors to bubble up, but we're not 100% sure if we need this yet to be honest. hence this being a draft still as we connect the full e2e dots together 😄

Reply:
i think i was imaging something like:

- function expects entry Foo to have "slug" field, but it doesn't
- so it returns false as the `result` & includes error details something like: `No "slug" field to validate on entry. Field not present in ContentType "blogPost"` or whatever which could be useful elsewhere like the logs? 🤔

Reply:
The example makes sense 👍🏻 The question is who takes care of logging that, is that the filter function itself or do you want to put that concern on the system invoking the function?

For context: The `errors` prop in the `graphql.query` response comes from the fact that we simply follow the [GQL spec](https://spec.graphql.org/October2021/#sec-Errors) and specifically this part:

> If a field error is raised, execution attempts to continue and a **partial result** is produced (see [Handling Field Errors](https://spec.graphql.org/October2021/#sec-Handling-Field-Errors)). The data entry in the response must be present. The errors entry should include all raised field errors.